### ISSUE TITLE
False positive: "Incompatible return value type"

### ISSUE THREAD

**Issue by jdufresne** [python/mypy#10059]
```
$ mypy --version
mypy 0.800
```

Input file:

```py
from typing import Union

def f(v: Union[int, str]) -> str:
    if isinstance(v, str):
        try:
            v = int(v)
        except ValueError:
            # return as is
            return v
    return str(v)
```

Output:

```
$ mypy --strict ~/test.py
/home/jon/test.py:9: error: Incompatible return value type (got "Union[int, str]", expected "str")
Found 1 error in 1 file (checked 1 source file)
```

Expected: No error. All return statments can only return `str`.
